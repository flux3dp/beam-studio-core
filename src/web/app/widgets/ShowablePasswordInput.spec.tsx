import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ShowablePasswordInput from './ShowablePasswordInput';

describe('test ShowablePasswordInput', () => {
  test('should render correctly', () => {
    const wrapper = shallow(<ShowablePasswordInput
      id="password-input"
      placeholder="Password"
    />);
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.find('img').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();

    const mockStopPropagation = jest.fn();
    wrapper.find('input#password-input').simulate('keyDown', {
      stopPropagation: mockStopPropagation,
    });
    expect(mockStopPropagation).toHaveBeenCalledTimes(1);
  });
});
