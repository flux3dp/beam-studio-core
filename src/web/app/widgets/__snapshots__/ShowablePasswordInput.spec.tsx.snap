// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ShowablePasswordInput should render correctly 1`] = `
<div
  className="showable-password-input"
>
  <input
    id="password-input"
    onKeyDown={[Function]}
    placeholder="Password"
    type="password"
  />
  <img
    onClick={[Function]}
    src="img/right-panel/icon-eyeopen.svg"
  />
</div>
`;

exports[`test ShowablePasswordInput should render correctly 2`] = `
<div
  className="showable-password-input"
>
  <input
    id="password-input"
    onKeyDown={[Function]}
    placeholder="Password"
    type="text"
  />
  <img
    onClick={[Function]}
    src="img/right-panel/icon-eyeclose.svg"
  />
</div>
`;
